### ISSUE TITLE
Contract Spec: Add CLI command that generates a TypeScript type from .wasm.

### ISSUE THREAD

**Issue by leighmcculloch** [stellar/soroban-tools#8]
We should:
- [x] In js-stellar-base, add `xdr.ScVal.toJSON(val, spec)`, and `xdr.ScVal.fromJSON(json, spec)`, inspired by the strval.rs code in this repo
  - Having this in js-stellar-base allows people to use the conversions even without the client binding generation.
- [x] Add typescript generation in rs-soroban-sdk, similar to how we generate rust, using the above js-stellar-base conversions to do parsing.
  - Similar to https://github.com/stellar/rs-soroban-sdk/pull/881
- [x] Add a `soroban contract bindings --wasm contract.wasm --output typescript > contract.ts` 
- [ ] Future: Research other libraries in other ecosystems and figure out a light-weight, but usable way to parse transaction results based on the spec as well. Write a follow-up issue for this.
(updated 2022-04-06)

**Comment by paulbellamy** [comment]
Blocked on https://github.com/stellar/rs-stellar-contract-sdk/issues/168

**Comment by smac711** [comment]
@paulbellamy is this still blocked and if so. What is the plan for unblocking it? 

Does this require a priority update? 
Is this required for Meridian?

**Comment by paulbellamy** [comment]
It was blocked on https://github.com/stellar/rs-stellar-contract-sdk/issues/168, which is closed now, so this is now unblocked.

I'd say it is completion or a nice-to-have for meridian. Not strictly necessary, though.

**Comment by paulbellamy** [comment]
And actually, in the platform sprint @tamirms had marked it as in-progress so I updated that in the soroban sprint to match.

**Comment by ire-and-curses** [comment]
DX nice-to-have

**Comment by paulbellamy** [comment]
https://github.com/stellar/rs-soroban-sdk/pull/881

**Comment by paulbellamy** [comment]
Related: I did a little prototype on generating typescript directly from `xdrgen` in https://github.com/stellar/xdrgen/tree/typescript, but it needs some underlying library support for serialization.

**Comment by Shaptic** [comment]
@paulbellamy this will be closed by stellar/rs-soroban-sdk#871 then #644 afaict, correct?

**Comment by paulbellamy** [comment]
Yes!
